src/Psr/Tracing/RelayOpenTelemetry.php

```diff
@@ -70,9 +70,13 @@ public function __construct(callable $client, ?TracerProviderInterface $tracerPr
      */
     public function __call(string $name, array $arguments)
     {
+        $stmt = $name . ' ' . implode(' ', $arguments);
+        $stmt = substr($stmt, 0, 1000);
+
         $span = $this->tracer->spanBuilder('Relay::' . strtolower($name))
             ->setAttribute('db.operation', strtoupper($name))
             ->setAttribute('db.system', 'redis')
+            ->setAttribute('db.statement', $stmt)
```

Comment:
We could use a one-liner using `sprintf()` here, however I'm somewhat worries that this will absolutely destroy the memory on some systems, because people sometimes write large amounts of data to Redis and it would all sit in PHP.

This would also need to be added to the calls lower in this class.

Reply:
Another issue we haven't tackled is calls to functions like `getMode()` or `getOption`. They probably shouldn't be traced.

Reply:
>write large amounts of data to Redis and it would all sit in PHP.

I am obviously with fine with not doing this at all, but it works with go-redis and this a plugin so the risks are low.

And I am probably wasting your time by sending one-line patches that require lengthy discussion so... just let me know what you decide :)

>This would also need to be added to the calls lower in this class.

Right, but that brings question - why do we need those methods at all? For example, I've tried to remove `scan` and got pretty much the same trace... am I missing something?

Reply:
>Right, but that brings question - why do we need those methods at all? For example, I've tried to remove scan and got pretty much the same trace... am I missing something?

Try iterating over more than the `$count` and you'll see the reference won't work.

Reply:
>I am obviously fine with not doing this at all, but it works with go-redis and this a plugin so the risks seems to be low.
>And I am probably wasting your time by sending one-line patches that require lengthy discussion so... just let me know what you decide :)

Your PR requires a lengthy discussion, because not all `__calls()` are Redis commands, some are local methods that aren't related to Redis 🙂

Reply:
>Try iterating over more than the $count and you'll see the reference won't work.

Indeed. Still don't understand what is happening though :shrug: